src/main/java/apoc/schema/Schemas.java

```diff
@@ -117,6 +135,101 @@ private AssertSchemaResult createUniqueConstraint(Schema schema, String lbl, Str
         return new AssertSchemaResult(lbl, key).unique().created();
     }
 
+    public List<AssertSchemaResult> recreateIndexes(long allIndexesTimeout, long dropAllIndexesTimeout, long recreateAllIndexesTimeout) throws ExecutionException, InterruptedException, IllegalArgumentException, TimeoutException {
+        Schema schema = db.schema();
+        List<AssertSchemaResult> result = Collections.synchronizedList(new ArrayList<>());
+        ExecutorService executorService = Pools.SINGLE;
+        List<IndexDefinition> indexesToRecreate = Util.inTxFuture(executorService, db, () -> {
+            List<IndexDefinition> idx = new ArrayList<>();
```

Comment:
can we extract these inner closures into methods so that we can just call them as a method (handle)

or simplify them more into a chain:

1. collect indexes + constraints into AssertResults
2. go over them and drop them
3. go over them and recreate them